src/monfaction.h

```diff
@@ -0,0 +1,41 @@
+#ifndef MONFACTION_H
+#define MONFACTION_H
+
+#include "string_id.h"
+#include "int_id.h"
+#include <unordered_map>
+
+class monfaction;
+class JsonObject;
+
+using mfaction_id = int_id<monfaction>;
+using mfaction_str_id = string_id<monfaction>;
+
+enum mf_attitude {
+    MFA_BY_MOOD = 0,    // Hostile if angry
+    MFA_NEUTRAL,        // Neutral even when angry
+    MFA_FRIENDLY        // Friendly
+};
+
+typedef std::unordered_map< mfaction_id, mf_attitude > mfaction_att_map;
+
+namespace monfactions
+{
+void finalize_monfactions();
+void load_monster_faction(JsonObject &jo);
+mfaction_id get_or_add_faction( const std::string &name );
+}
+
+class monfaction {
+    public:
+        mfaction_id id;
+        mfaction_id base_faction;
+        mfaction_str_id name;
```

Comment:
(I know it was already that way) This is confusing, all the other objects use `loadid` for the integer id and `id` for the string id and `name` for the actual name (that is shown to the user and translated).

Even through monster factions are not shown to the user (and currently have no member for a proper name), this should be made consistent at one time.